src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -143,6 +150,13 @@ public function start()
             throw new \RuntimeException('Failed to start the session');
         }
 
+        if (null !== $this->emulateSameSite) {
+            $originalCookie = SessionUtils::popSessionCookie(session_name(), session_id());
+            if (null !== $originalCookie) {
+                header(sprintf('%s; SameSite=%s', $originalCookie, $this->emulateSameSite));
```

Comment:
The code cannot use `header()` here. It would break many scenarios. The only part of the code that can output something is the Response class.

Reply:
This same method also uses `session_start`, which also sends headers that are not sent through the `Response` class but directly. So this change isn't doing anything the method wasn't doing already.

I do agree that adding a `Cookie` to the `Reponse` would be much nicer, but I can't think of a reasonable way to get it there, seeing there is no `Response` available this early in the application lifecycle.

Reply:
@rpkamp is right, this is how sessions work for now...